src/BenchmarkDotNet/Horology/TimeUnit.cs

```diff
@@ -1,29 +1,32 @@
 ﻿using System.Linq;
+using BenchmarkDotNet.Encodings;
 
 namespace BenchmarkDotNet.Horology
 {
     public class TimeUnit
     {
-        public string Name { get; }
+        private readonly MultiEncodingString name;
+        
+        public MultiEncodingString Name => name;
         public string Description { get; }
         public long NanosecondAmount { get; }
 
-        private TimeUnit(string name, string description, long nanosecondAmount)
+        private TimeUnit(MultiEncodingString name, string description, long nanosecondAmount)
         {
-            Name = name;
+            this.name = name;
             Description = description;
             NanosecondAmount = nanosecondAmount;
         }
 
         public TimeInterval ToInterval(long value = 1) => new TimeInterval(value, this);
 
-        public static readonly TimeUnit Nanosecond = new TimeUnit("ns", "Nanosecond", 1);
-        public static readonly TimeUnit Microsecond = new TimeUnit("us", "Microsecond", 1000);
-        public static readonly TimeUnit Millisecond = new TimeUnit("ms", "Millisecond", 1000 * 1000);
-        public static readonly TimeUnit Second = new TimeUnit("s", "Second", 1000 * 1000 * 1000);
-        public static readonly TimeUnit Minute = new TimeUnit("m", "Minute", Second.NanosecondAmount * 60);
-        public static readonly TimeUnit Hour = new TimeUnit("h", "Hour", Minute.NanosecondAmount * 60);
-        public static readonly TimeUnit Day = new TimeUnit("d", "Day", Hour.NanosecondAmount * 24);
+        public static readonly TimeUnit Nanosecond = new TimeUnit(new MultiEncodingString("ns", "ns"), "Nanosecond", 1);
+        public static readonly TimeUnit Microsecond = new TimeUnit(new MultiEncodingString("\u03BCs", "us"), "Microsecond", 1000);
+        public static readonly TimeUnit Millisecond = new TimeUnit(new MultiEncodingString("ms", "ms"), "Millisecond", 1000 * 1000);
+        public static readonly TimeUnit Second = new TimeUnit(new MultiEncodingString("s", "s"), "Second", 1000 * 1000 * 1000);
+        public static readonly TimeUnit Minute = new TimeUnit(new MultiEncodingString("m", "m"), "Minute", Second.NanosecondAmount * 60);
+        public static readonly TimeUnit Hour = new TimeUnit(new MultiEncodingString("h", "h"), "Hour", Minute.NanosecondAmount * 60);
+        public static readonly TimeUnit Day = new TimeUnit(new MultiEncodingString("d", "d"), "Day", Hour.NanosecondAmount * 24);
```

Comment:
It would be nice to have an additional constructor for `MultiEncodingString` which takes only one argument. Also, it would be nice to have implicit/explicit conversions from string to `MultiEncodingString`.